build-system/babel-plugins/babel-plugin-transform-jss/index.js

```diff
@@ -0,0 +1,139 @@
+ * Takes a .jss.js file and transforms the `useStyles` export to remove side effects
+ * and directly return the classes map.
+ *
+ * @example
+ * In:
+ * ```
+ * import {createUseStyles} from 'react-jss'
+ *
+ * const JSS = { button: { fontSize: 12 }}
+ * export const CSS = ''
+ * export const useStyles = createUseStyles(JSS);
+ * ```
+ *
+ * Out:
+ * ```
+ * const JSS = { button: { fontSize: 12 }}
+ * const classes = {button: 'button-1'}
+ * export const useStyles = () => classes;
+ * export const CSS = `button-1 { fontSize: 12 }`
+ * ```
+ */
+
+const {create} = require('jss');
+const {default: preset} = require('jss-preset-default');
+
+module.exports = function ({types: t}) {
+  function isJssFile(filename) {
+    return filename.endsWith('.jss.js');
+  }
+
+  function compileJss(JSS) {
+    const jss = create();
+    jss.setup(preset());
+    return jss.createStyleSheet(JSS);
+  }
+
+  const isIdent = (path, ident) => path.node.id && path.node.id.name === ident;
+  const sheetMap = new WeakMap();
+
+  function findAndCompileJss(path, state) {
+    if (sheetMap.has(state.file)) {
+      return;
+    }
+
+    //TODO: @jridgewell, is there a better way to find module scope?
+    const moduleScopePath = path.findParent((p) => p.parent.type === 'File');
+    // Try to find the createUseStyles. The first arg will be the JSS.
+    // Compile the JSS and place into the sheetMap.
+    moduleScopePath.traverse({
+      CallExpression(path) {
+        if (path.node.callee.name !== 'createUseStyles') {
+          return;
+        }
+
+        const {confident, value: JSS} = path.get('arguments')[0].evaluate();
+        if (!confident) {
+          throw new Error(
+            `First argument to createUseStyles must be statically evaluatable.`
+          );
+        }
+        sheetMap.set(state.file, compileJss(JSS));
+      },
+    });
+    if (!sheetMap.has(state.file)) {
+      throw new Error(`Could not find createUseStyles.`);
+    }
+  }
+
+  return {
+    visitor: {
+      VariableDeclarator(path, state) {
+        // TODO: Can I skip the whole file if not jss?
+        const {filename} = state.file.opts;
+        if (!isJssFile(filename)) {
+          return;
+        }
+        findAndCompileJss(path, state);
+
+        if (isIdent(path, 'useStyles')) {
+          // Convert jss classes object to ast equivalent.
+          const classesVal = t.objectExpression(
+            Object.entries(sheetMap.get(state.file).classes).map(([k, v]) =>
+              t.objectProperty(t.identifier(k), t.stringLiteral(v))
+            )
+          );
+          path
+            .getStatementParent()
+            .insertBefore(
+              t.variableDeclaration('const', [
+                t.variableDeclarator(t.identifier('classes'), classesVal),
+              ])
+            );
+          path
+            .get('init')
+            .replaceWith(
+              t.arrowFunctionExpression([], t.identifier('classes'))
+            );
+          path.stop();
```

Comment:
Calling `path.stop` will stop all transforms. At most, you want to use `path.skip()`, but I don't think even that's necessary.